stdlib/LinearAlgebra/src/matmul.jl

```diff
@@ -452,23 +488,48 @@ Base.@constprop :aggressive function generic_matmatmul_wrapper!(C::StridedMatrix
         return _rmul_or_fill!(C, β)
     end
     matmul2x2or3x3_nonzeroalpha!(C, tA, tB, A, B, α, β) && return C
-    # We convert the chars to uppercase to potentially unwrap a WrapperChar,
-    # and extract the char corresponding to the wrapper type
-    tA_uc, tB_uc = uppercase(tA), uppercase(tB)
     alpha, beta = promote(α, β, zero(T))
-    if alpha isa Union{Bool,T} && beta isa Union{Bool,T}
-        if tA_uc == 'S' && tB_uc == 'N'
-            return BLAS.symm!('L', tA == 'S' ? 'U' : 'L', alpha, A, B, beta, C)
-        elseif tA_uc == 'N' && tB_uc == 'S'
-            return BLAS.symm!('R', tB == 'S' ? 'U' : 'L', alpha, B, A, beta, C)
-        elseif tA_uc == 'H' && tB_uc == 'N'
-            return BLAS.hemm!('L', tA == 'H' ? 'U' : 'L', alpha, A, B, beta, C)
-        elseif tA_uc == 'N' && tB_uc == 'H'
-            return BLAS.hemm!('R', tB == 'H' ? 'U' : 'L', alpha, B, A, beta, C)
-        end
+    blasfn = _valtypeparam(val)
+    if alpha isa Union{Bool,T} && beta isa Union{Bool,T} && blasfn ∈ (BlasFlag.SYMM, BlasFlag.HEMM)
+        _blasfn = blasfn
+        αβ = (alpha, beta)
+    else
+        _blasfn = BlasFlag.NONE
+        αβ = (α, β)
+    end
+    _symm_hemm_generic!(C, tA, tB, A, B, αβ..., Val(_blasfn))
+    return C
+end
+function _lrchar_ulchar(tA, tB)
```

Comment:
Do we want to aggressively const-propagate here?

Reply:
Actually, the constant propagation seems to work in this function even without the aggressive propagation hint.